### PR TITLE
Downintegrate changes from the OpenMRNIDF repository

### PR DESCRIPTION
**Pull request by balazsracz**
This PR pulls in changes that happened in the OpenMRNIDF repository. Most of these were authored by Mike Dunston.

- All instances of ifdef ESP32 replaced with ESP_PLATFORM
- Removed conditional code for supporting ESP-IDF v3.x and v4.x
  Only ESP-IDF v5 and up is now supported
- new feature for OPENMRN_HAVE_SOCKET_FSTAT
- fixed compile errors around spiram
- refactored definition of ADC pin into a separate class in a separate file
- fixes around printf PRIu32.
- do not print error messages on CAN overflow
- fix to not start the hub twice in the wifi manager
- Remove deprecated hardware can driver
- Adds I2C driver written by Mike Dunston for the ESP32.

### CODE REVIEW

**Review by atanisoft — APPROVED** [review]

**Comment by atanisoft** [comment]
Thanks @balazsracz for taking this on, I had a local branch to do similarly but held off on pushing it upstream due to various commitments.

One thing to note, we should publish a "final" OpenMRNLite version prior to merge since arduino-esp32 2.x has a hard dependency on IDF v4.4.x and arduino-esp32 3.x is still under development using IDF v5.1.x. We might consider bumping the OpenMRNLite version when the IDF v5.x dependency version has been released.

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx [hunk 1]
```diff
@@ -876,9 +876,11 @@ void* twai_watchdog(void* param)
...
             LOG(INFO,
                 "ESP-TWAI: "
-                "RX:%d (pending:%zu,overrun:%d,discard:%d,missed:%d,lost:%d) "
-                "TX:%d (pending:%zu,suc:%d,fail:%d) "
-                "Bus (arb-loss:%d,err:%d,state:%s)",
+                "RX:%" PRIu32 " (pending:%zu,overrun:%" PRIu32
+                ",discard:%" PRIu32 ",missed:%" PRIu32 ",lost:%" PRIu32 ") "
+                "TX:%" PRIu32 " (pending:%zu,suc:%" PRIu32
+                ",fail:%" PRIu32 ") "
+                "Bus (arb-err:%" PRIu32 ",err:%" PRIu32 ",state:%s)",
```
Comment (atanisoft) [inline src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:883]: https://github.com/atanisoft/OpenMRNIDF/pull/7/files#diff-88ee088b2bbc83933050861edc298188dbc20d3dbc31bf2fa003d9ae7c989603 has changed all of the types for these values from uint32_t to size_t which cleans up this log statement considerably.

File: src/freertos_drivers/esp32/Esp32Ledc.cxx [hunk 2]
```diff
@@ -35,10 +35,6 @@
 // Ensure we only compile this code for the ESP32 family of MCUs.
 #if defined(ESP32)
```
Comment (atanisoft) [inline src/freertos_drivers/esp32/Esp32Ledc.cxx:36]: missed update from ESP32 -> ESP_PLATFORM

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx [hunk 3]
```diff
@@ -37,12 +37,9 @@
 
 // Ensure we only compile this code for the ESP32 family of MCUs and that the
 // ESP-IDF version is supported for this code.
```
Comment (atanisoft) [inline src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:39]: we can truncate this comment after MCUs.

File: src/openlcb/Bootloader.hxx [hunk 4]
```diff
@@ -43,7 +43,7 @@
 #include <string.h>
 #include <unistd.h>
 
-#ifdef ESP32
+#ifdef ESP_PLATFORM
 #include "bootloader_hal.h"
 #else
 #include "freertos/bootloader_hal.h"
```
Comment (atanisoft) [inline src/openlcb/Bootloader.hxx:49]: I think we should modify libify.sh to create the freertos directory so we can consolidate these include paths, I think we had discussed this but haven't implemented a fix yet

File: src/freertos_drivers/esp32/Esp32Gpio.hxx [hunk 5]
```diff
@@ -36,27 +36,16 @@
...
-
-// esp_rom_gpio.h is a target agnostic replacement for esp32/rom/gpio.h
-#if __has_include(<esp_rom_gpio.h>)
 #include <esp_rom_gpio.h>
-#elif ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(4,0,0)
-#include <rom/gpio.h>
-#else
-#include <esp32/rom/gpio.h>
-#endif
+#include <soc/adc_channel.h>
```
[the remark this answers is not in the document]
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32Gpio.hxx:46]: done

[the remark this answers is not in the document]
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32Gpio.hxx:46]: I will also remove the ADC macro from blow.

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx [hunk 6]
```diff
@@ -716,7 +714,7 @@ static void twai_isr(void *arg)
 {
     BaseType_t wakeup = pdFALSE;
     uint32_t events = twai_hal_get_events(&twai.context);
-    ESP_EARLY_LOGV(TWAI_LOG_TAG, "events: %08x", events);
+    ESP_EARLY_LOGV(TWAI_LOG_TAG, "events: %04x", events);
```
[the remark this answers is not in the document]
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:717]: done

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx (hunk 1 above)
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:883]: Let's do that in a different round. I have not reviewed the 5.1.0 changes yet.

File: src/freertos_drivers/esp32/Esp32Ledc.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32Ledc.cxx:36]: done

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx (hunk 3 above)
Comment (balazsracz) [reply src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:39]: done

File: src/openlcb/Bootloader.hxx (hunk 4 above)
Comment (balazsracz) [reply src/openlcb/Bootloader.hxx:49]: Filed #778

File: src/freertos_drivers/esp32/Esp32HardwareTwai.cxx (hunk 1 above)
Comment (atanisoft) [reply src/freertos_drivers/esp32/Esp32HardwareTwai.cxx:883]: That's fine, the bulk of the changes you have already pulled in, it's mostly further cleanup and adjustments for 5.1/5.2 and those are minimilar IIRC

**Pull request merged**